ArcGISRuntimeSDKQt_CppSamples/EditData/SnapGeometryEdits/SnapGeometryEdits.cpp

```diff
@@ -0,0 +1,335 @@
+  m_lineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor(255, 45, 0), 2, this);
+  m_polygonSymbol = new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor(255, 0, 0, 75),
+                      new SimpleLineSymbol(SimpleLineSymbolStyle::Dash, QColor(0, 0, 0), 1.0, this), this);
+}
+
+void SnapGeometryEdits::createConnections()
+{
+  // Allow user to edit existing graphics by clicking on them
+  connect (m_mapView, &MapQuickView::mouseClicked, this, [this](const QMouseEvent& mouseEvent)
+  {
+    if (!m_geometryEditor->isStarted())
+    {
+      m_mapView->identifyGraphicsOverlayAsync(m_graphicsOverlay, mouseEvent.position(), 10 ,false).then(this, [this](IdentifyGraphicsOverlayResult* result)
+      {
+        // Handle editing selected graphics, if any
+        auto identifyResult = std::unique_ptr<IdentifyGraphicsOverlayResult>(result);
+        // Return if no graphics were identified
+        if (identifyResult->graphics().isEmpty())
+          return;
+
+        m_editingGraphic = identifyResult->graphics().first();
+
+        // Hide the graphic currently being edited
+        m_editingGraphic->setVisible(false);
+
+        // Start the geometry editor with the graphic's geometry
+        m_geometryEditor->start(m_editingGraphic->geometry());
+
+        emit geometryEditorStartedChanged();
+
+        result->deleteLater();
+      });
+    }
+    emit canUndoChanged();
+    emit elementIsSelectedChanged();
+  });
+
+  // Enable or disable buttons when mouse is released (ie after a drag operation)
+  connect(m_mapView, &MapQuickView::mouseReleased, this, [this](const QMouseEvent&)
+  {
+    emit canUndoChanged();
+    emit elementIsSelectedChanged();
+  });
+}
+
+bool SnapGeometryEdits::geometryEditorStarted() const
+{
+  return (m_geometryEditor && m_geometryEditor->isStarted());
+}
+
+bool SnapGeometryEdits::canUndo()
+{
+  return (m_geometryEditor && m_geometryEditor->canUndo());
+}
+
+bool SnapGeometryEdits::elementIsSelected()
+{
+  return (m_geometryEditor && m_geometryEditor->selectedElement() && m_geometryEditor->selectedElement()->canDelete());
+}
+
+// Toggles snapping using the enabled state from the snap settings
+void SnapGeometryEdits::snappingEnabledStatus(bool snappingCheckedState)
+{
+  m_geometryEditor->snapSettings()->setEnabled(snappingCheckedState);
+}
+
+// Toggles snapping for the point layer snap source at [index] using the checked value from the snap settings
+void SnapGeometryEdits::pointSourceEnabledStatus(bool snappingCheckedState, int index)
+{
+  m_pointSourceCheckedState[index] = snappingCheckedState;
+  m_pointSourceList[index]->setEnabled(m_pointSourceCheckedState[index]);
+
+}
+
+// Toggles snapping for the polyline layer snap source at [index] using the checked value from the snap settings
+void SnapGeometryEdits::polylineSourceEnabledStatus(bool snappingCheckedState, int index)
```

Comment:
why do we need to separate points and lines? can we share the logic/structures for all geometry types?